### ISSUE TITLE
FW: Portland- SAB Code Assignment

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#727]
In GitLab by @kdjstudios on Mar 1, 2018, 10:22

**Submitted by:** "Martin Villegas" <martin.villegas@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-03-01-82997

**Server:** Internal

**Client/Site:** Portland

**Account:** NA

**Issue:** 

We have a code in Portland and Stockton that I need to remove (Carry Forward BaseRate 9990) but I can’t. below is Cori’s approval.

(Attached the full email)

![image001__1_](/uploads/af951eb7de2b05eb87edf8a8330879b1/image001__1_.jpg)

[original_message__2_.html](/uploads/cedd81289a884528e07f7e0fa9df5970/original_message__2_.html)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 1, 2018, 10:22

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 1, 2018, 10:24

My reply:

>  If a code's name is Greyed out it is already deactivated in the system and can not be assigned to accounts. Due to historical reports we can not 'delete' codes. Please let us know if you have any questions. 

Martin's response:

> I am still able to use the code, please see screenshot below. Juan in West Memphis tried removing the word true next to it and wrote false and the code disappeared, we just want to make sure we are safe to do that?
> ![image001__2_](/uploads/4c202736ba77b20153858fdf5e37e1dd/image001__2_.jpg)

My response:

> Thank you for the update. We have informed the support team, as you should not have been able to use the code after it was deactivated. In regards to changing the true to false, I believe those are not supposed to be visible to users either, and have informed the support team to look into this and we will follow up with their responses accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 5, 2018, 02:56

@kdjstudios 

>> Juan in West Memphis tried removing the word true next to it and wrote false and the code disappeared, we just want to make sure we are safe to do that?

**That would not be right, we will have the true/false column removed in next release.**

![image](/uploads/f471da23ca57509e144832ad50efc23f/image.png)

**Fees which don't have any action button against it are those Fees which system used to generate certain reports or invoice calculation. It would be better if we don't tamper with those.** For eg: `7778 Bad Debt` is used for bad debt report. 

Additionally, we would be removing True/False column as it's not required. Greyed-out fee is enough to indicate whether it's activated or deactivated.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Mar 5, 2018, 06:24

@smasih, 

Please also check the issue related to Human read-able report. Perhaps its a new switch - TeleScan and we might require more details to set it up for human readable. @tim.traylor has created a new ticket #730 for this. This switch might be converted to VCC in coming months any ways.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 5, 2018, 09:40

@smasih @abakshi 

Update sent to martin:

> The support team has informed me that users should not have access to the true/false column; and that editing those values will effect historical reports, invoices, and prevent other items in SA Billing from working correctly. They are working to insure that this column will not be visible to users. 
> 
> Also we would request to know exactly which item was edited in the true/false column so the support team can return it to the needed value.  Once it has been returned to the correct state, the support team will then start to look into why a disabled fee was showing in your drop down lists.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 5, 2018, 14:17

Martin replied:

>  The Code was 9990 Carry Forward BaseRate for West Memphis.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 6, 2018, 04:01

Hi [Kyle] ( @kdjstudios ),   

> Also we would request to know exactly which item was edited in the true/false column so the support team can return it to the needed value.  Once it has been returned to the correct state, the support team will then start to look into why a disabled fee was showing in your drop-down lists.

* The functionality of only **available** additional fees should be shown on staged fees is totally intact. We have successfully tested on Merge. So, it means the process was not completed on its first step.  
As I checked the additional fees **Carry Forward BaseRate** is still **available**. So, please make it disable first.  

As [Sumeet]  ( @smasih ) mentioned about the true/false column that it should not be visible to anyone. We have successfully changed the option on Merge. Please check. 

The disabled additional fees & activities are grayed out.  

**Note** - Some additional fees are pre-defined which cannot be disabled, those are also grayed out.
  

Hope this helps. 

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 6, 2018, 04:01

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 6, 2018, 06:26

@pchaudhary  made the required changes live on merge.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 6, 2018, 10:05

@smasih Can you please check and tell us if the 9990 code was ever used in Portland and Stockton?  If so, please let us know the last date and account it was on?  

If the codes were never used, please go ahead and remove them as system defined codes.

Thx

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 7, 2018, 24:02

Hi [Tim] ( @tim.traylor ),   
We have checked for the code **9990** and below are our findings - 
1.  Stockton, CA - It was never used after  **Sep 17, 2014**
1.  Portland -  It was never used after **Mar 26, 2015**  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 7, 2018, 24:11

@pchaudhary please deactivate the code in Portland as per the original request.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 7, 2018, 03:57

@kdjstudios @tim.traylor   
Should we remove the code **9990** as the system defined code?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 7, 2018, 14:37

Update sent out:

> Thank you for the update. The support team believes they were able to revert the changes made. 
> 
> The support team checked the first two sites mentioned, Stockton and Portland, and found the following on last usage of the code.  If you are needing to also know other sites last usage of the code please specify the site names.
> 
> -  Stockton, CA - It was last used in Sep 17, 2014
> -  Portland -  It was last used in Mar 26, 2015
> 
> By removing these historical codes it will directly effect all reports, historical data, accounts, etc that used or referenced that code from those dates prior. There will be no way to undo the removal once it has completed.  We want to confirm both you and Cori's knowledge of the effects and approval before the support team removes the codes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 7, 2018, 14:37

@smasih @pchaudhary 

Once we have approval from Operations I will let you know.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 7, 2018, 22:06

@kdjstudios   
Thanks for the update. We'll wait for the response.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 8, 2018, 06:05

@kdjstudios 

We have hidden the `true/False` column.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 8, 2018, 08:32

@smasih @pchaudhary 

Thanks for the update and hiding the true/false column to prevent future issues. I am still waiting to hear back from Cori on approval to remove codes.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 21, 2018, 05:15

Hi [Kyle] ( @kdjstudios ), 

We are still waiting for Cori's approval to remove the codes. 

However, we have successfully hidden the True/False column from the fields. Please check & mark as ready if all goes well on your end too so that we can release this in the Upcoming release.


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2018, 10:14

@pchaudhary 

Thank you, I have sent another update out:

> Is anything else needed on this ticket? We have not received a response in about two weeks and will be marking this as closed if we do not receive any updates.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2018, 14:54

Cori responded:

> Hi All- It's ok- you need to remove the System codes. Thanks! Cori

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 21, 2018, 17:02

Please remove the codes from the following sites:

Portland
Stockton
West Memphis

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 21, 2018, 23:31

Hi [Kyle] ( @kdjstudios ), 

Thanks for the update.  



> Cori responded:
>
> > Hi All- It's ok- you need to remove the System codes. Thanks! Cori


* We have successfully disabled the code *9990*.  

  ![image](/uploads/e98f906a8345ebc7142db37054d0d1d9/image.png)




* Now, we are ready to release the code for hiding the True/False column from all of the fields. Please check & approve accordingly. 


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 22, 2018, 08:51

@pchaudhary 

Just to confirm the True/False column (Sr Management) is only visible to those with that security level access correct?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 22, 2018, 23:14

Hi [Kyle] ( @kdjstudios ), 


> Just to confirm the True/False column (Sr Management) is only visible to those with that security level access correct?

* Sorry, but I am bit confused here. Will you please explain that which True/False column (Sr Management) are you asking about? 

* One thing I need to mention here that as per the developed code, no one has access to see the column in which the True/False is mentioned. However, the **red button** through which the User can activate/de-activate the fees still has the same functionality as previous.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 23, 2018, 08:16

@pchaudhary 

Thanks for the update, I am referring to the Senior Management (SnrMngmt) setting in the security levels
- None: This means the columns are not visible
- R: This means the columns are visible but not able to be edited
- RW: this means the columns are visible and able to be edited

I was under the impression that this setting, is the one that controls the "Available" Column in like the site edit page for example. 

- I do want to mention, that since I have been a (admin) user, I have always had access to the "Available" column in SA Billing. I just have never changed any of the settings. Which is why I was under the impression above that it is related to the Senior Management.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 23, 2018, 10:45

Hi [Kyle] ( @kdjstudios ),  


Thanks for the update.  
I would like to mention here as previous, that we only hid the *True/False* text column to all of the fees from the Site edit page.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 23, 2018, 10:49

Martin replied:

> Stockton still has the code.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 23, 2018, 13:27

Hi [Kyle] ( @kdjstudios ), 

> Martin replied:
>
> > Stockton still has the code.

* Sorry, my bad. I think I mistakenly did that on Merge. I have successfully disabled the code in production, now. Please ask Martin to check & confirm.


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 26, 2018, 08:50

I responded:

> This should be update now. May you please confirm. 

Martin Replied:

> Yes, its removed.

Cori Replied:

> Hi- I replied last week to please not delete the system codes. Thanks Cori

Cori Replied again:

> I left  a word out of this reply... It's ok. You DON'T need to remove the system codes. Cori

I replied:

> Per your approval last week the support team has already deleted those codes; as we had warned prior to asking for your approval that there would be no way to undo these changes.
>
> If needed, we can see if the support team is able to miraculously revert those changes. 

Cori Replied:

> Or add them back in...

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 27, 2018, 24:07

Hi [Kyle] ( @kdjstudios ),   

On Additional fee there is an `additional check` in the backend.

Which make the fee system defined or regular fee.

System-defined fee cannot be deactivated from GUI. While the regular one can be.

So What I meant by removed was that we have disabled the `System-defined` check box, which made it like a  regular additional fee.

In short, the fee is only disabled not deleted.


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 27, 2018, 08:02

@pchaudhary 

Per Cori's request, please re enable the fees then.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 27, 2018, 23:46

Hi [Kyle] ( @kdjstudios ),  


We have successfully re-enabled the `9990` code. Please confirm with Cori & mark it for close if all meet her requirements.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 28, 2018, 08:34

@pchaudhary 

Thank you, I have informed Operations and will follow up with their response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 29, 2018, 12:03

Marking for close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2018, 13:19

closed
